### DISCUSSION TITLE
[Core Protocol] How to add new preinstall contract into protocol

### DISCUSSION THREAD

**Discussion by skye240** [ethereum-optimism/developers#395] [Core Protocol]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

How to add new contract as preinstall/predeploy into new chain base on OP Stack?

**Comment by hash1go** [comment]
Hello. There likely isn't a standard method for adding pre-deployed contracts other than rewriting the code.
Could you share any methods you have actually tried?

**Reply by skye240** [reply]
First step, I want to understand the flow code to add new pre-deployed/preinstall. After that I able to build tool that help other develop able to add new easy into op stack

**Comment by Chomtana** [comment]
It seem like predeploys are specified in the genesis.json but I don't pretty sure.

OP Stack is going to have a big migration in genesis deployment (The doc is still using an obsoleted `sync()` method with `go run cmd/main.go genesis l2`)

But with this commit 3 weeks ago, it seems like they are moving away from both `sync()` and `go run cmd/main.go genesis l2`

https://github.com/ethereum-optimism/optimism/commits/bde6a96e65d812f33681069009cdf3fa4f1f87d7/packages/contracts-bedrock/src/libraries/Preinstalls.sol

Tagging @sbvegan for a help in discussing with protocol engineer team about recent genesis.json generation procedure and confirm if predeploys are actually located in genesis.json.

**Comment by joohhnnn** [comment]
This may give a help now :https://github.com/ethereum-optimism/developers/discussions/769#discussioncomment-12587215

**Comment by opfocus** [comment]
If there is already an answer you find acceptable, you can mark it as the accepted answer (you can also mark your own answer as accepted), and then close the discussion. You can also close it without marking any answer. Additionally, you can continue to add comments.
